src/main/java/training/endava/skeleton/Main.java

```diff
@@ -41,7 +59,96 @@ public static void main(String argv[]){
 
         personRepository.findAll().forEach(System.out::println);
 
+        */
 
+        // 1
+        List<String> companyNameList = companies.stream()
+                .map(Company::getName)
+                .collect(Collectors.toList());
+        System.out.println(companyNameList);
+
+        // 2
+        companies.stream()
+                .map(Company::getFoundationYear)
+                .mapToInt(Integer::intValue)
+                .sorted()
+                .forEach(System.out::println);
+
+        // 3
+        Company oldest = companies.stream()
+                .min((c1, c2) -> Integer.compare(c1.getFoundationYear(), c2.getFoundationYear()))
+                .get();
+
+        System.out.println(oldest);
+
+        // 4
+        int count = companies.stream()
+                .map(Company::getPersonList)
+                .map(List::size)
+                .reduce(0,(s1,s2) -> s1+s2)
+                .intValue();
+        System.out.println(count);
+
+        // 5
+        Person p99 = companies.stream()
+                .map(Company::getPersonList)
+                .flatMap(List<Person>::stream)
+                .reduce(new Person(), (per1, per2) -> {per1.setName(per1.getName() + per2.getName()); return per1;});
```

Comment:
Have you verified the resulted name? per1.getName() for the identity value will be null, so the resulted person will have the name starting with "null". You should initialize the name with empty string in the constructor.